salt/states/junos.py

```diff
@@ -188,20 +181,19 @@ def rollback(name, id, **kwargs):
 
     """
     ret = {"name": name, "changes": {}, "result": True, "comment": ""}
-    ret["changes"] = __salt__["junos.rollback"](id, **kwargs)
+    ret["changes"] = __salt__["junos.rollback"](**kwargs)
     return ret
 
 
 @resultdecorator
-def diff(name, d_id):
+def diff(name, **kwargs):
```

Comment:
Changing an interface for a function which has been published without a deprecation cycle having been done is not allowed. You could extend the interface

    def diff(name, d_id, **kwargs) 

and do a versionadded, but the interface has to be backwards compatible unless a deprecation cycle is performed

Reply:
same as above, also to add

For diff, the documentation clearly stated that it expects "id" as the parameter name but the name of the argument was "d_id". This is a bug and we don't see a reason to support "d_id". Also, this will go inline to changes done at https://github.com/saltstack/salt/commit/93ee5ee2b01b6f752f5538d0da413719d410d65c#diff-15ad308a12ea5b6d5188dd4a94eab4a0R490

Reply:
see reply for salt/states/junos.py about deprecation cycle and changing interfaces
The label of a positional argument doesn't really matter here compared to a documentation typo/error, but the positional argument is required otherwise the interface is changing without a deprecation cycle for the interface

Reply:
@dmurphy18 I have reverted to the old code function definition itself.